File: Domain/Entities/Order.cs
using System;
using System.Collections.Generic;

namespace Domain.Entities
{
    public class Order
    {
        public int Id { get; set; }
        public bool IsPayed { get; set; }
        public DateTime OrderDate { get; set; }
        public string StreetShipping { get; set; }
        public string HouseNrShipping { get; set; }
        public string HouseBusShipping { get; set; }
        public string PostalcodeShipping { get; set; }
        public string CityShipping { get; set; }
        public int CustomerId { get; set; }
        public Customer Customer { get; set; }
        public ICollection<OrderLine> OrderLines { get; set; }
    }
}

File: Presentation_Angular/ClientApp/src/app/Shared/models/order.ts
import { Customer } from "./customer";
import { ShoppingCartItem } from "./shoppingCartItem";

export interface Order {
  id?: number;
  isPayed?: boolean;
  orderDate?: Date;
  streetShipping: string;
  houseNrShipping: string;
  houseBusShipping: string;
  postalcodeShipping: string;
  cityShipping: string;
  customerId: number;
  customer?: Customer;
  orderLines: ShoppingCartItem[];
  orderSummary?: {
    totalCount: number;
    totalExVat: number;
    vat: number;
    totalInclVat: number;
  };
}
